protocol/consistencychecks.go

```diff
@@ -225,43 +229,65 @@ func (cc *ConsistencyChecks) verifySTR(str *m.SignedTreeRoot) error {
 	// verify hash chain
 	// TODO: clearly verify which is the client's actual expectation.
 	// See #81
-	if (str.Epoch == cc.CurrentEpoch && bytes.Equal(cc.SavedSTR, str.Signature)) ||
-		(str.Epoch == cc.CurrentEpoch+1 && str.VerifyHashChain(cc.SavedSTR)) {
+	if (str.Epoch == cc.SavedSTR.Epoch && bytes.Equal(cc.SavedSTR.Signature, str.Signature)) ||
+		(str.Epoch == cc.SavedSTR.Epoch+1 && str.VerifyHashChain(cc.SavedSTR.Signature)) {
 		return nil
 	}
+
+	// TODO: verify the directory's policies as well. See #115
 	return ErrorBadSTR
 }
 
-// verifySignedPromise checks if the returned response is match
-// with the directory's policy. Then it calls verifyFulfilledPromise
-// to verify whether issued TBs was inserted in the directory
-// as promised. Finally, it verifies the returned promise
-// based on the request type. Note that the directory returns a promise
-// iff the returned proof is a proof of absence.
+// verifyFulfilledPromise verifies issued TBs was inserted
+// in the directory as promised.
+func (cc *ConsistencyChecks) verifyFulfilledPromise(uname string,
+	str *m.SignedTreeRoot,
+	ap *m.AuthenticationPath) error {
+	if !cc.useTBs {
+		return nil
+	}
+
+	for name, tb := range cc.TBs {
```

Comment:
I'm still confused about this loop. I feel like it's doing two different, unrelated operations: one is checking that the TB for a given `uname` is properly included in the next epoch, and the other is clearing old TBs (which will be taken care of once we address #116). While we don't have #116, I think we should have separate functions for these two operations. 

To verify the fulfilled promise, I don't think the client should have to loop through the TB map; it should only operate on `tb, ok := cc.TBs[uname]`.